distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/ServerCli.java

```diff
@@ -29,21 +29,30 @@
 
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.List;
 import java.util.Locale;
 
 /**
  * The main CLI for running Elasticsearch.
  */
 class ServerCli extends EnvironmentAwareCommand {
 
+    interface ServerProcessPostExitProcessor {
+        void process(int exitCode);
+
+        void waitForCompletion() throws InterruptedException;
```

Comment:
Does this need to throw `InterruptedException` as part of its contract? There's nothing sensible the caller can do, other than just exit...

Reply:
No it does not, I can remove it